web/packages/teleterm/src/ui/DocumentGatewayKube/DocumentGatewayKube.tsx

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+import { Flex, Text, ButtonPrimary } from 'design';
+import { useAsync } from 'shared/hooks/useAsync';
+
+import Document from 'teleterm/ui/Document';
+import * as types from 'teleterm/ui/services/workspacesService';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import { useWorkspaceContext } from 'teleterm/ui/Documents';
+import { DocumentTerminal } from 'teleterm/ui/DocumentTerminal';
+import { connectToKube } from 'teleterm/ui/services/workspacesService';
+import { retryWithRelogin } from 'teleterm/ui/utils';
+import { routing } from 'teleterm/ui/uri';
+
+export const DocumentGatewayKube = (props: {
```

Comment:
The overall approach with `DocumentGatewayKube` is fine, however this component doesn't need to follow exactly what `DocumentGatewayCliClient` does.

In general, `DocumentGatewayCliClient` expects that another document is going to create the gateway and thus it doesn't attempt to do so itself. For `DocumentGatewayKube`, you don't need `WaitingForGateway` with all that timeout logic and so on.

What I think we could do here is take the `const [connectAttempt, createGateway] = useAsync` part from `WaitingForGateway` and move it up to `DocumentGatewayKube`. Then we could also take `useEffect` and remove that timeout-related stuff from there.

Then when it comes to what this component is actually supposed to render, we could make a switch on `connectAttempt`. If it's `''` or `'processing'`, I think we could simply return `<Document visible={visible} px={2}>` with nothing inside – the doc status already handles showing a progress bar.

If it's `'error'`, we need to show that error somewhere and a retry button which executes `createGateway` on click. I'd take the Flex from `WaitingForGatewayContent` and replace its content with this part from `DocumentTerminal/Reconnect`

https://github.com/gravitational/teleport/blob/69702eea4f2d587d3ad6fb534b78ee8588ebdcf8/web/packages/teleterm/src/ui/DocumentTerminal/Reconnect.tsx#L42-L60

`message` could be replaced with, idk, "Could not create a gateway.". The button could say just "Retry".

If the attempt status is `'success'`, then all that's left is to return `DocumentTerminal`.